openfisca_aotearoa/variables/regulation/student_allowance/interpretation.py

```diff
@@ -0,0 +1,122 @@
+
+
+class student_allowance__is_living_with_a_parent(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = """living with a parent has the same meaning as in section 3(1) of the Social Security Act 1964"""
+
+
+class student_allowance__is_married_or_partnered(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "married or partnered as per Student Allowances Regulations 1998"
+    reference = """
+        https://legislation.govt.nz/regulation/public/1998/0277/latest/whole.html#DLM259903
+        (a) means having a spouse; and
+        (b) for the avoidance of doubt, does not include a person who is legally married
+            but who does not have a spouse (as that term is defined in this subclause)
+        """
+
+    def formula(persons, period, parameters):
+        return persons('student_allowance__has_a_spouse', period)
+
+
+class student_allowance__has_a_supported_child(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "has a supported child as per Student Allowances Regulations 1998"
+    reference = """
+    https://legislation.govt.nz/regulation/public/1998/0277/latest/whole.html#DLM259968
+    supported child, in relation to a student applying for an allowance, means a person younger than 24—
+        (a)
+
+        whose well-being and financial support are the responsibility of the student; and
+        (b) who lives with that student at least half of the time; and
+        (c) who is not in receipt of—
+            (i) an allowance continued by regulation 3(a) to (c); or
+            (ii) jobseeker support, sole parent support, an emergency benefit, or a supported living payment under the Social Security Act 1964; or
+            (iii) New Zealand superannuation under the New Zealand Superannuation and Retirement Income Act 2001 or a veteran’s pension under
+                the Veterans’ Support Act 2014; or
+            (iv) payments under any government-assisted scheme (other than the Student Loan Scheme) which, in the opinion of the chief executive,
+                is similar to a benefit under the Social Security Act 1964; or
+            (v) income before tax from employment or self-employment which exceeds $80 per week; and
+        (d) in respect of whom no orphan’s benefit or unsupported child’s benefit is payable under the Social Security Act 1964
+    """
+
+
+class student_allowance__partner_has_a_supported_child(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "their spouse has a supported child, as per Student Allowances Regulations 1998"
+
+
+class student_allowance__has_a_spouse(Variable):
+    value_type = bool
+    entity = Person
+    definition_period = MONTH
+    label = "Has spouse as per Student Allowances Regulations 1998"
+    reference = """
```

Comment:
I think the reference link is all that's required, we can lose the repeated text form the act